File: app/src/main/java/ua/dp/rename/dniprostreets/core/SnappyInitializer.java
package ua.dp.rename.dniprostreets.core;

import ua.dp.rename.dniprostreets.repo.SnappyMigrator;
import ua.dp.rename.dniprostreets.repo.SnappyRepository;

public class SnappyInitializer implements ComponentInitializer {

    SnappyRepository db;

    public SnappyInitializer(SnappyRepository db) {
        this.db = db;
    }

    @Override
    public void init() {
        new SnappyMigrator(db).tryMigration();
        db = null;
    }
}

File: app/src/main/java/ua/dp/rename/dniprostreets/core/SnappyInitializer.kt
package ua.dp.rename.dniprostreets.core

import ua.dp.rename.dniprostreets.repo.SnappyMigrator
import ua.dp.rename.dniprostreets.repo.SnappyRepository

class SnappyInitializer(
      private var db: SnappyRepository?
) : ComponentInitializer {

   override fun init() {
      SnappyMigrator(db).tryMigration()
      db = null
   }
}
